### PR TITLE
BINDINGS/GO: Ucp listener functionality.

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What
BINDINGS/GO:: Ucp listener functionality.

### CODE REVIEW

File: bindings/go/src/ucx/utils.go [hunk 1]
```diff
@@ -24,3 +33,88 @@ func CBytes(data string) unsafe.Pointer {
...
+		// IPv6 address.
+		ip6 := a.IP.To16()
+		if ip6 == nil {
+			return nil, &net.AddrError{Err: "non-IPv6 address", Addr: a.IP.String()}
+		}
+		sa.ss_family = C.AF_INET6
+		var sin6 *C.struct_sockaddr_in6 = (*C.struct_sockaddr_in6)(unsafe.Pointer(sa))
+		sin6.sin6_port = C.htons(C.ushort(a.Port))
+		charBytes := C.CString(ip6.String())
+		C.inet_pton(C.AF_INET, charBytes, unsafe.Pointer(&sin6.sin6_addr))
```
Comment (yosefe) [inline bindings/go/src/ucx/utils.go:96]: AF_INET6

File: bindings/go/src/ucx/utils.go [hunk 2]
```diff
@@ -24,3 +33,88 @@ func CBytes(data string) unsafe.Pointer {
...
+		// "0.0.0.0" or "::", means the entire IP addressing
+		// space. For some historical reason, it is used to
+		// specify "any available address" on some operations
+		// of IP node.
+		//
+		// When the IP node supports IPv4-mapped IPv6 address,
+		// we allow a listener to listen to the wildcard
+		// address of both IP addressing spaces by specifying
+		// IPv6 wildcard address.
+		if len(a.IP) == 0 || a.IP.Equal(net.IPv4zero) {
```
Comment (yosefe) [inline bindings/go/src/ucx/utils.go:83]: IPv6zero?

File: bindings/go/tests/listener_test.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+	flushReq1, _ := ep.FlushNonBlocking(nil)
+	defer flushReq1.Close()
+	for connReq == nil {
+		worker1.Progress()
+		worker2.Progress()
+	}
+
+	backwardEpParams := (&UcpEpParams{}).SetConnRequest(connReq)
+	backwardEpParams.SetPeerErrorHandling().SetErrorHandler(logErrorHandler)
```
Comment (yosefe) [inline bindings/go/tests/listener_test.go:76]: "backwardEp" means "peerEp"/"replyEp"?

File: bindings/go/tests/listener_test.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	ucpParams := (&UcpParams{}).EnableTag()
+
+	ucpWorkerParams := (&UcpWorkerParams{}).SetThreadMode(UCS_THREAD_MODE_SINGLE)
+
+	listenerParams := &UcpListenerParams{}
+	listenerParams.SetSocketAddress(addr)
+
+	epParams := &UcpEpParams{}
+	var logErrorHandler UcpEpErrHandler = func(ep *UcpEp, status UcsStatus) {
+		t.Logf("Error handler called with status %d", status)
```
Comment (yosefe) [inline bindings/go/tests/listener_test.go:28]: should it be fatal?

File: bindings/go/tests/listener_test.go (hunk 3 above)
Comment (petro-rudenko) [reply bindings/go/tests/listener_test.go:76]: It's ep created from connectionRequest on a server.

File: bindings/go/src/ucx/utils.go (hunk 2 above)
Comment (petro-rudenko) [reply bindings/go/src/ucx/utils.go:83]: https://github.com/golang/go/blob/d9349175ad13ab228d3224c4bff8e3e1b2936b42/src/net/ipsock_posix.go#L169
> // In general, an IP wildcard address, which is either
	// "0.0.0.0" or "::", means the entire IP addressing
	// space. For some historical reason, it is used to
	// specify "any available address" on some operations
	// of IP node.
	//
	// When the IP node supports IPv4-mapped IPv6 address,
	// we allow a listener to listen to the wildcard
	// address of both IP addressing spaces by specifying
	// IPv6 wildcard address.

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@petro-rudenko pls squash

**Pull request merged**